Salesforce 10.5 (#1089)
mulesoft/docs-connectors: salesforce/10.5/modules/ROOT/pages/salesforce-connector-reference.adoc

| Include null values a| Boolean |  Include null values in the Apex Rest API request. This parameter applies to the Invoke Apex Rest Method operation only. |  false |
|===

==== Connection Types
[[sfdc-config_basic]]
===== Basic Authentication

====== Parameters
[%header%autowidth.spread]
|===
| Name | Type | Description | Default Value | Required
| Connection Timeout a| Number |  Specifies the amount of time that the consumer attempts to establish a connection before it times out. Works in tandem with Time unit. Values less than one millisecond are converted to 0. |  0 |
| Time unit a| Enumeration, one of:

** NANOSECONDS
** MICROSECONDS
** MILLISECONDS
** SECONDS
** MINUTES
** HOURS
** DAYS |  Time unit to use for the Connection Timeout parameter |  SECONDS |
| Proxy configuration a| <<ProxyConfiguration>> |  Proxy configuration for the connector. |  |
| Object TTL a| Number |  Specifies the amount of time an object in the Connections pool lives if it was not used. |  30 |
| Time unit a| Enumeration, one of:

** NANOSECONDS
** MICROSECONDS
** MILLISECONDS
** SECONDS
** MINUTES
** HOURS
** DAYS |  A time unit that qualifies the Object TTL |  SECONDS |

> | Max Entries a| Number |  Specifies the amount of objects that live in the pool at a specific moment. When this number is reached and a new object is needed, no new object will be created and the application waits until a connection is released. |  10 |

Reviewer: ```suggestion
| Max Entries a| Number |  Specifies the maximum number of objects that live in the pool at a specific moment. When the maximum is reached and a new object is needed, no new object is created until a connection is released. |  10 |
```

|===
[[sfdc-config_config-with-oauth]]
===== OAuth v2.0

====== Parameters
[%header%autowidth.spread]
|===
| Name | Type | Description | Default Value | Required
| Connection Timeout a| Number |  Specifies the amount of time that the consumer attempts to establish a connection before it times out. Works in tandem with Time unit. Values less than one millisecond are converted to 0. |  0 |
| Time unit a| Enumeration, one of:

** NANOSECONDS
** MICROSECONDS
** MILLISECONDS
** SECONDS
** MINUTES
** HOURS
** DAYS |  Time unit to use for the Connection Timeout parameter |  SECONDS |
| Proxy configuration a| <<ProxyConfiguration>> |  Proxy configuration for the connector. |  |
| Object TTL a| Number |  Specifies the amount of time an object in the Connections pool lives if it was not used. |  30 |
| Time unit a| Enumeration, one of:

** NANOSECONDS
** MICROSECONDS
** MILLISECONDS
** SECONDS
** MINUTES
** HOURS
** DAYS |  Time unit that qualifies the Object TTL |  SECONDS |

> | Max Entries a| Number |  Specifies the amount of objects that live in the pool at a specific moment. When this number is reached and a new object is needed, no new objects will be created and the application waits until a connection is released. |  10 |

Reviewer: ```suggestion
| Max Entries a| Number |  Specifies the maximum number of objects that live in the pool at a specific moment. When the maximum is reached and a new object is needed, no new object is created until a connection is released. |  10 |
```

| TLS configuration a| <<Tls>> |  If HTTPS was configured as the protocol, then the user must configure the keystore configuration, at a minimum.  |  |
| Reconnection a| <<Reconnection>> |  When the application is deployed, a connectivity test is performed on all connectors. If set to true, deployment fails if the test doesn't pass after exhausting the associated reconnection strategy. |  |
| Consumer Key a| String |  The OAuth consumerKey as registered with the service provider |  | x
| Consumer Secret a| String |  The OAuth consumerSecret as registered with the service provider |  | x
| Authorization Url a| String |  The service provider's authorization endpoint URL |  `\https://login.salesforce.com/services/oauth2/authorize` |
| Access Token Url a| String |  The service provider's accessToken endpoint URL |  `\https://login.salesforce.com/services/oauth2/token`|
| Scopes a| String |  The OAuth scopes to request during the OAuth dance. If not provided, it defaults to the OAuth scopes in the annotation. |  |
| Resource Owner Id a| String |  The resourceOwnerId that each component should use if it doesn't reference otherwise. |  |

> | Before a| String |  The name of a flow to execute right before starting the OAuth dance |  |

Reviewer: ```suggestion
| Before a| String |  The name of a flow to execute immediately before starting the OAuth dance |  |
```

| After a| String |  The name of a flow to execute right after an accessToken has been received |  |

> | Listener Config a| String |  A reference to an `<HTTP:listener-config` /> to use to create the listener that catches the access token callback endpoint. |  | x

Reviewer: ```suggestion
| Listener Config a| String |  A reference to an `<HTTP:listener-config` /> to use to create the listener that receives the access token callback endpoint. |  | x
```

| Object Store a| String |  A reference to the object store that should be used to store each resource owner ID's data. If not specified, runtime will automatically provision the default one. |  |
|===
[[sfdc-config_jwt]]
===== OAuth JWT

====== Parameters
[%header%autowidth.spread]
|===
| Name | Type | Description | Default Value | Required
| Connection Timeout a| Number |  Specifies the amount of time that the consumer attempts to establish a connection before it times out. Works in tandem with Time unit. Values less than one millisecond are converted to 0. |  0 |
| Time unit a| Enumeration, one of:

** NANOSECONDS
** MICROSECONDS
** MILLISECONDS
** SECONDS
** MINUTES
** HOURS
** DAYS |  Time unit to use for the Connection Timeout parameter |  SECONDS |
| Proxy configuration a| <<ProxyConfiguration>> |  Proxy configuration for the connector. |  |
| Object TTL a| Number |  Specifies the amount of time an object in the Connections pool lives if it was not used. |  30 |
| Time unit a| Enumeration, one of:

** NANOSECONDS
** MICROSECONDS
** MILLISECONDS
** SECONDS
** MINUTES
** HOURS
** DAYS |  A time unit that qualifies the Object TTL |  SECONDS |

> | Max Entries a| Number |  Specifies the amount of objects that live in the pool at a specific moment. When this number is reached and a new object is needed, no new object will be created, and the application waits until a connection is released. |  10 |

Reviewer: ```suggestion
| Max Entries a| Number |  Specifies the maximum number of objects that live in the pool at a specific moment. When the maximum is reached and a new object is needed, no new object is created until a connection is released. |  10 |
```

> | Api Version a| Number |  The API version used. Defaults to the latest api version supported by the connector. |  |

Reviewer: ```suggestion
| Api Version a| Number |  The API version used. Defaults to the latest API version supported by the connector. |  |
```

|===
[[sfdc-config_oauth-user-pass]]
===== OAuth Username Password

====== Parameters
[%header%autowidth.spread]
|===
| Name | Type | Description | Default Value | Required
| Connection Timeout a| Number |  Specifies the amount of time that the consumer attempts to establish a connection before it times out. Works in tandem with Time unit. Values less than one millisecond are converted to 0. |  0 |
| Time unit a| Enumeration, one of:

** NANOSECONDS
** MICROSECONDS
** MILLISECONDS
** SECONDS
** MINUTES
** HOURS
** DAYS |  Time unit to use for the Connection Timeout parameter |  SECONDS |
| Proxy configuration a| <<ProxyConfiguration>> |  Proxy configuration for the connector. |  |
| Object TTL a| Number |  Specifies the amount of time, an object in the Connections pool will live if it was not used. |  30 |
| Time unit a| Enumeration, one of:

** NANOSECONDS
** MICROSECONDS
** MILLISECONDS
** SECONDS
** MINUTES
** HOURS
** DAYS |  A time unit that qualifies the Object TTL |  SECONDS |

> | Max Entries a| Number |  Specifies the amount of objects that live in the pool at a specific moment. When this number is reached and a new object is needed, no new object will be created and the application waits until a connection is released. |  10 |

Reviewer: ```suggestion
| Max Entries a| Number |  Specifies the maximum number of objects that live in the pool at a specific moment. When the maximum is reached and a new object is needed, no new object is created until a connection is released. |  10 |
```

Reviewer: I won't keep marking these. It only caught my eye because using "amount" instead of "number" is one of my pet peeves. It's not important.

| Max Entries a| Number |  Specifies the amount of objects that live in the pool at a specific moment. When this number is reached and a new object is needed, no new object will be created and the application waits until a connection is released. |  10 |
| Api Version a| Number |  The API version used. Defaults to the latest api version supported by the connector. |  |
| Consumer Key a| String |  Consumer key for the Salesforce-connected app |  | x
| Key Store a| String |  Path to the keystore used to sign data during authentication |  | x
| Store Password a| String |  Password of keystore |  | x
| Certificate Alias a| String |  Alias of the certificate |  |
| Principal a| String |  Username of desired Salesforce user to take action on behalf of. |  | x
| Token Endpoint a| String |  URL that points to the server responsible for providing the authentication token. According to Salesforce, it is `\https://login.salesforce.com/services/oauth2/token`. If implementation is for a community, it is `\https://acme.force.com/customers/services/oauth2/token` (where `acme.force.com/customers` is your community URL). |  `\https://login.salesforce.com/services/oauth2/token` |

> | TLS configuration a| <<Tls>> |  Defines a configuration for TLS, which can be used from both the client and server sides to secure communication for the Mule app. When using the HTTPS protocol, the HTTP communication is secured using TLS or SSL. If HTTPS is configured as the protocol then the user needs to configure at least the keystore in the `tls:context` child element of the `listener-config`. |  |

Reviewer: Here, the description for TLS configuration was updated to what we have on our spreadsheet, but it's not updated in other places.
